source/hash_table.c

```diff
@@ -53,7 +53,10 @@ static uint64_t s_hash_for(struct hash_table_state *state, const void *key) {
 }
 
 static size_t s_index_for(struct hash_table_state *map, struct hash_table_entry *entry) {
-    return (size_t)(entry - map->slots);
+    AWS_PRECONDITION(hash_table_state_is_valid(map));
+    size_t index = entry - map->slots;
+    AWS_POSTCONDITION(index < map->size);
```

Comment:
No need for `AWS_POSTCONDITION(hash_table_state_is_valid(map))`?

Reply:
Done